### PR TITLE
feat(ADR): Documentation Strategy (#129)

### PR DESCRIPTION
**Pull request by SenorMonito**
This ADR proposes a documentation strategy for 4 specific use cases for documentation and the proposed structure using GitHub as the Content Management System.

### CODE REVIEW

**Review by PaddyMc — APPROVED** [review]
:+1:

File: docs/architecture/adr-002-docs-structure.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+##### docs/Explanation/articles
+
+This SHALL BE a folder that contains a sub-folder for each published article. By published this COULD REFER to blog posts. The folder should be name such that it describes the article purpose. Each sub-folder SHALL CONTAIN all the content relevant to the article (e.g. images, bibliographies etc). These articles can be converted into pdf using pandoc. 
+
+In order to do this
+
+- There SHOULD BE a makefile with targets for calling pandoc - Note: the process for building pdf are not part of the commit or release processes, but ad-hoc
+- There SHOULD BE a latex template file that can create pdf files with consistent look and feel. This COULD BE the [Eisvogel template](https://github.com/Wandmalfarbe/pandoc-latex-template) with suitable modifications
+- The makefile and template should be independent of the article
+- THere SHOULD BE a README.md that describes how to use and build articles
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:193]:
```suggestion
- There SHOULD BE a README.md that describes how to use the makefile and template and build articles
```

File: docs/architecture/adr-002-docs-structure.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+> **Note:** Explanations can come in other forms, particularly issue discussion and Pull Requests.
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:195]: did you mean for that `>` symbol to be here? Not sure what it does

File: docs/architecture/adr-002-docs-structure.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+#### docs/Tutorials
+
+As indicated in the overview, these SHALL BE documents that target beginners and will take a user step-by-step through a process with the aim of achieving some goal. Please see the [Divio tutorial page](https://documentation.divio.com/tutorials/) for detail.
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:199]:
```suggestion
As indicated in the overview, tutorials SHALL BE documents that target beginners and guide a user step-by-step through a process with the aim of achieving some goal. Please see the [Divio tutorial page](https://documentation.divio.com/tutorials/) for details.
```

File: docs/architecture/adr-002-docs-structure.md [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+
+- There SHALL BE a folder for each tutorial - see the [Cosmos SDK tutorials](https://github.com/cosmos/sdk-tutorials) as an example
+- The folder SHALL CONTAIN all the content relevant for that tutorial. 
+- The content SHOULD BE consistent in format with [SDK tutorials](https://tutorials.cosmos.network/).
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:203]:
```suggestion
- There SHALL BE a folder for each tutorial. See the [Cosmos SDK tutorials](https://github.com/cosmos/sdk-tutorials) as an example.
- The folder SHALL CONTAIN all of the content that is relevant for that tutorial. 
- The content SHOULD BE consistent in format with [Cosmos SDK tutorials](https://tutorials.cosmos.network/). 
```

File: docs/architecture/adr-002-docs-structure.md [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+
+As indicated in the overview, these SHALL BE documents that target beginners and will take a user step-by-step through a process with the aim of achieving some goal. Please see the [Divio tutorial page](https://documentation.divio.com/tutorials/) for detail.
+
+- There SHALL BE a folder for each tutorial - see the [Cosmos SDK tutorials](https://github.com/cosmos/sdk-tutorials) as an example
+- The folder SHALL CONTAIN all the content relevant for that tutorial. 
+- The content SHOULD BE consistent in format with [SDK tutorials](https://tutorials.cosmos.network/). 
+
+#### docs/How-To
+
+Compared to tutorials, [How-Tos](https://documentation.divio.com/how-to-guides/) are help an experienced reader solve a specific problem. These SHALL USE templates similar to the tutorials - see above.
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:207]:
```suggestion
In contrast to tutorials, [how-to guides](https://documentation.divio.com/how-to-guides/) are a series of actionable steps to help an experienced reader solve a specific problem. These how-to guides SHALL USE templates similar to the tutorials - see above.
```

File: docs/architecture/adr-002-docs-structure.md [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+
+### Templates
+
+Documentation SHOULD USE templates for documentation such as module messages etc.
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:211]:
```suggestion
The documentation SHOULD USE Markdown templates to develop structured technical content, like module messages follow templates in the Cosmos SDK. 
```

File: docs/architecture/adr-002-docs-structure.md [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+
+Documentation SHOULD USE templates for documentation such as module messages etc. 
+
+- [The good docs project](https://github.com/thegooddocsproject)
+- [Readme editor](https://readme.so/editor)
+
+
+#### Code Comments
+
+These are included as comments also form part of documentation. Comments SHALL FOLLOW recommendation as per [Conventional Comments](https://conventionalcomments.org/)
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:219]:
```suggestion
PR review comments also form part of the documentation. Comments SHALL FOLLOW recommendation as per [Conventional Comments](https://conventionalcomments.org/)
```

Comment (barriebyron) [reply docs/architecture/adr-002-docs-structure.md:219]: do we mean comments in the PR review process? or what are we talking about here?

File: docs/architecture/adr-002-docs-structure.md [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+<label> [decoration]: <subject>
+
+[discussion]
+```
+
+where `label = (praise|nitpick|suggestion|issue|question|thought|chore)`
+
+#### Commit Comments
+
+Commits comments will also follow a similar format as laid out following standard defined by [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary). This SHOULD BE enforced as part of [pre-commit](https://pre-commit.com/) checks
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:231]:
```suggestion
Commits comments will also follow a similar format as laid out following the standard defined by [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary). This commit convention SHOULD BE enforced as part of [pre-commit](https://pre-commit.com/) checks.
```

File: docs/architecture/adr-002-docs-structure.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+
+## Consequences
+
+This section describes the resulting context, after applying the decision. 
+
+### Backwards Compatibility
+
+Once this ADR is implemented existing documentation will be migrated from existing sources that include:
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:239]:
```suggestion
After this ADR is implemented, existing documentation will be migrated from existing sources that include:
```

File: docs/architecture/adr-002-docs-structure.md [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+
+- Notion
+- Other Git repos
+- Published papers
+- Blog posts 
+
+### Positive
+
+As a result of this
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:248]:
```suggestion
As a result of this documentation strategy:
```

File: docs/architecture/adr-002-docs-structure.md [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+
+- Content will follow best practice that will be easy to navigate and read
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:250]:
```suggestion
- Content development and maintenance will follow best practices that ensure content is easy to navigate and read
```

File: docs/architecture/adr-002-docs-structure.md [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+- Content will be in a consistent format
+- Commits, Issues and Pull Requests will follow best practice
```
Comment (barriebyron) [inline docs/architecture/adr-002-docs-structure.md:252]:
```suggestion
- Commits, Issues, and Pull Requests in the repo will follow best practices
```

**Review by barriebyron — APPROVED** [review]
@SenorMonito I love this formal approach! In past jobs, we have adhered to 3 primary content types: concept (called Explanation by Divio), task (how-to), and reference. Tutorials were managed by another group since tutorials are closely tied to software release versions and typically include assets required to run code for that particular release. 

This documentation strategy is a wonderful step in a good direction. Be proud!

I provided more editorial suggestions, but went ahead and approved so you can merge it after you commit the changes you agree with. Cheerio!

**Comment by barriebyron** [comment]
food for thought @SenorMonito 
Perhaps add some language to add flexibility to achieve the best outcome, following https://www.digitalocean.com/community/tutorials/digitalocean-s-technical-writing-guidelines

File: docs/architecture/adr-002-docs-structure.md [hunk 13]
```diff
@@ -0,0 +1,270 @@
+# ADR 002: Documentation Structure
+
+## Status
+
+PROPOSED
```
Comment (SenorMonito) [inline docs/architecture/adr-002-docs-structure.md:5]:
```suggestion
ACCEPTED
```

**Review by SenorMonito — COMMENTED** [review]
Based on reviewer feedback. moving to accepted status

**Pull request merged**